src/mlpack/tests/test_tools.hpp

```diff
@@ -35,4 +34,9 @@
 
 #endif
 
+// Require the approximation L to be within a relative error of E respect to the
+// actual value R.
+#define REQUIRE_RELATIVE_ERR( L, R, E ) \
+    BOOST_REQUIRE_LE( abs((R) - (L)), (E) * (R))
```

Comment:
This will fail if R is negative.  I think maybe a better choice here is to use the Boost test tools implementation... by default `BOOST_REQUIRE_CLOSE()` checks both the condition above and also | R - L | <= E * |L|, which works for nearest neighbor search, but not furthest neighbor search (unless you, as an argument, set `E = e(1 + e)`).  So you could do that, or you could look at `boost/test/test_tools.hpp` and see how `BOOST_REQUIRE_CLOSE` is defined, and you can develop a new macro that uses `boost::test_tools::check_is_close` with the `FPC_WEAK` comparison type, and I think that would work for what you are doing.